src/Platforms/Echo.Platforms.AsmResolver/Emulation/IFunctionMarshaller.cs

```diff
@@ -0,0 +1,26 @@
+using AsmResolver.DotNet;
+
+namespace Echo.Platforms.AsmResolver.Emulation;
+/// <summary>
+/// Provides methods for marshalling functions into pointers
+/// </summary>
+public interface IFunctionMarshaller
+{
+    /// <summary>
+    /// Gets the machine the marshaller is targeting.
+    /// </summary>
+    CilVirtualMachine Machine
+    {
+        get;
+    }
+
+    /// <summary>
+    /// Returns pointer for method descriptor
+    /// </summary>
+    nuint GetFunctionPointer(IMethodDescriptor method);
```

Comment:
We cannot use `nint` or `nuint` for pointer-typed values, as the host process may have a different bitness than the emulated process. This is why `ulong` is used for addresses throughout the library instead.